census_view.cpp

```diff
@@ -932,6 +936,844 @@ inline std::vector<std::string> const& CensusViewDataViewModel::all_headers() co
+
+    return type;
+}
+
+wxString CensusViewTable::GetColLabelValue(int col)
+{
+    if(col == Col_CellNum)
+        {
+        return "Cell";
+        }
+
+    auto const& header = all_headers()[view_.visible_columns_[col - 1]];
+    return insert_spaces_between_words(header);
+}
+
+bool CensusViewTable::AppendRows(size_t numRows)
+{
+    auto grid = GetView();
+    LMI_ASSERT(grid != nullptr);
+
+    wxGridTableMessage msg
+        (this
+        ,wxGRIDTABLE_NOTIFY_ROWS_APPENDED
+        ,bourn_cast<int>(numRows)
+        );
+    grid->ProcessTableMessage(msg);
+
+    return true;
+}
+
+bool CensusViewTable::DeleteRows(size_t pos, size_t num_rows)
+{
+    auto grid = GetView();
+    LMI_ASSERT(grid != nullptr);
+
+    auto const ipos = bourn_cast<int>(pos);
+    auto inum_rows = bourn_cast<int>(num_rows);
+    auto const cur_num_rows = GetView()->GetNumberRows();
+    if(ipos >= cur_num_rows)
+        {
+        wxFAIL_MSG(wxString::Format
+            (wxT("Called CensusViewTable::DeleteRows(pos=%d, N=%d)\nPos value is invalid for present table with %d rows")
+            ,ipos
+            ,inum_rows
+            ,cur_num_rows
+            )
+        );
+
+        return false;
+        }
+
+    if(inum_rows > cur_num_rows - ipos)
+        {
+        inum_rows = cur_num_rows - ipos;
+        }
+
+    if(grid)
+        {
+        wxGridTableMessage msg
+            (this
+            ,wxGRIDTABLE_NOTIFY_ROWS_DELETED
+            ,ipos
+            ,inum_rows
+            );
+        grid->ProcessTableMessage(msg);
+        }
+
+    return true;
+}
+
+bool CensusViewTable::AppendCols(size_t numCols)
+{
+    auto grid = GetView();
+    LMI_ASSERT(grid != nullptr);
+
+    wxGridTableMessage msg
+        (this
+        ,wxGRIDTABLE_NOTIFY_COLS_APPENDED
+        ,bourn_cast<int>(numCols)
+        );
+    grid->ProcessTableMessage(msg);
+
+    return true;
+}
+
+bool CensusViewTable::DeleteCols(size_t pos, size_t num_cols)
+{
+    auto grid = GetView();
+    LMI_ASSERT(grid != nullptr);
+
+    auto const ipos = bourn_cast<int>(pos);
+    auto inum_cols = bourn_cast<int>(num_cols);
+    auto const cur_num_cols = GetView()->GetNumberCols();
+
+    if(ipos >= cur_num_cols)
+        {
+        wxFAIL_MSG(wxString::Format
+            (wxT("Called CensusViewTable::DeleteCols(pos=%d, N=%d)\nPos value is invalid for present table with %d cols")
+            ,ipos
+            ,inum_cols
+            ,cur_num_cols
+            )
+        );
+
+        return false;
+        }
+
+    if(inum_cols > cur_num_cols - ipos)
+        {
+        inum_cols = cur_num_cols - ipos;
+        }
+
+    if(grid)
+        {
+        wxGridTableMessage msg
+            (this
+            ,wxGRIDTABLE_NOTIFY_COLS_DELETED
+            ,ipos
+            ,inum_cols
+            );
+        grid->ProcessTableMessage(msg);
+        }
+
+    return true;
+}
+
+void CensusViewTable::SetFirstColumnReadOnly()
+{
+    if(CanHaveAttributes())
```

Comment:
Why do we need to check for this? Can this ever be false? And, even if it can, why not call `SetColAttr()` anyhow (it just does nothing in this case)?